Negotiation/FormatNegotiator.php

```diff
@@ -73,31 +72,27 @@ public function getBest($header, array $priorities = [])
                 }
                 continue;
             }
-            if ($options['prefer_extension']) {
-                // ensure we only need to compute $extensionHeader once
-                if (!isset($extensionHeader)) {
-                    if (preg_match('/.*\.([a-z0-9]+)$/', $request->getPathInfo(), $matches)) {
-                        $extension = $matches[1];
-                    }
 
+            if (isset($options['prefer_extension']) && $options['prefer_extension'] && !isset($extensionHeader)) {
+                $extension = pathinfo($request->getPathInfo(), PATHINFO_EXTENSION);
+
+                if (!empty($extension)) {
                     // $extensionHeader will now be either a non empty string or an empty string
-                    $extensionHeader = isset($extension) ? (string) $request->getMimeType($extension) : '';
+                    $extensionHeader = $request->getMimeType($extension);
                     if ($header && $extensionHeader) {
-                        $extensionHeader = ','.$extensionHeader;
+                        $header .= ',';
                     }
-                }
-                if ($extensionHeader) {
                     $header .= $extensionHeader.'; q='.$options['prefer_extension'];
                 }
             }
 
-            $mimeTypes = $this->normalizePriorities(empty($priorities) ? $options['priorities'] : $priorities);
+            $mimeTypes = empty($priorities) ? $options['priorities'] : $priorities;
```

Comment:
We should maybe create an issue / a pr for symfony 2.8 or 3.0 to add a new method ``Request::getMimeTypes()`` and see what the symfony staff think about this.